packages/api/src/interfaces.ts

```diff
@@ -18,17 +18,16 @@ export type ApiClientResponse<
   E extends Exclude<HttpStatusCode, HttpSuccessCodes> = Exclude<HttpStatusCode, HttpSuccessCodes>,
 > =
   | {[K in keyof S]: ApiClientSuccessResponse<K, S[K]>}[keyof S]
-  | {[K in E]: ApiClientSErrorResponse<K>}[E]
-  | ApiClientSErrorResponse<HttpStatusCode.INTERNAL_SERVER_ERROR>;
+  | {[K in E]: ApiClientErrorResponse<K>}[E]
+  | ApiClientErrorResponse<HttpStatusCode.INTERNAL_SERVER_ERROR>;
 
 export type ApiClientResponseData<T extends ApiClientResponse> = T extends {ok: true; response: infer R} ? R : never;
 
-export type GenericRequestObject = Record<string, unknown>;
-export type GenericResponseObject = {code: (code: number) => void};
+export type GenericOptions = Record<string, unknown>;
 
 export type ServerApi<T extends Record<string, APIClientHandler>> = {
   [K in keyof T]: (
-    ...args: [...args: Parameters<T[K]>, req?: GenericRequestObject, res?: GenericResponseObject]
+    ...args: [...args: Parameters<T[K]>, opts?: GenericOptions]
```

Comment:
I removed req / res but added generic opts instead, we have API implementations that pass additional options, like `publishBlock`. Might not be a bad thing in general to allow this as API implementation are called internally as well.

https://github.com/ChainSafe/lodestar/blob/19b723c55042090bcb0442f83002ca129e37817d/packages/beacon-node/src/api/impl/beacon/blocks/index.ts#L202